src/core/task.ts

```diff
@@ -102,7 +102,13 @@ export class Task<T> {
    * with the given backoff strategy.
    */
   retry(backoff: Backoff, maxRetries: number): Task<T> {
-    return this._retry(backoff, maxRetries, 0);
+    if (maxRetries < 0) {
+      throw new Error('assertion failed: number of task retries must be greater than or equals zero');
```

Comment:
I thought about that when writing it, but line of thought was that if you pass an invalid argument you would want it to fail asap.